src/Programs.tsx

```diff
@@ -0,0 +1,326 @@
+import React, { ReactElement, useState, SetStateAction, Dispatch, useEffect, useContext } from 'react';
+import { Button, Table, Header, Grid, Segment } from 'semantic-ui-react';
+import { setUserProgram, retrievePrograms } from './Database';
+import { NumericOnlyInput } from './components';
+import { NewProgramContext } from './Context';
+
+export interface Program {
+  name: string;
+  author: string;
+  level: string;
+  days: string;
+  program: Array<Array<{ exerciseName: string; Sets: number; Reps: number }>>;
+  notes: string;
+  setViewedName: Dispatch<SetStateAction<string>>;
+  setEditing: Dispatch<SetStateAction<boolean>>;
+}
+
+type ProgramNoMethods = Omit<Program, 'setViewedName' | 'setEditing'>;
+
+function Programs(): ReactElement {
+  const [viewedName, setViewedName] = useState<string>('');
+  const [editing, setEditing] = useState<boolean>(false);
+  const [database] = useState<Array<ProgramNoMethods>>(retrievePrograms());
+  const [rows] = useState<Array<ReactElement>>(() => {
```

Comment:
Same, I have never seen anyone use JSX and React elements in the state

Reply:
fixed (?)